packages/traceloop-sdk/traceloop/sdk/tracing/tracing.py

```diff
@@ -108,7 +109,64 @@ def __new__(
             if propagator:
                 set_global_textmap(propagator)
 
-            init_instrumentations()
+            instrument_set = False
+            if not instruments:
+                init_instrumentations()
+                instrument_set = True
+            else:
+                if 'openai' in instruments:
```

Comment:
I suggest we use an enum for this (it can be an enum with string values), so the user can choose values from an enum instead of guessing whether we called it `openai` or `OpenAI`.